Gateway/Response/PaymentPosCloudHandler.php

```diff
@@ -74,6 +74,42 @@ public function handle(array $handlingSubject, array $paymentResponse)
         // do not send order confirmation mail
         $payment->getOrder()->setCanSendNewEmailFlag(false);
 
+        if (!empty($paymentResponse['Response']) &&
+            !empty($paymentResponse['Response']['AdditionalResponse'])
+        ) {
+            $pairs = explode('&', $paymentResponse['Response']['AdditionalResponse']);
+
+            foreach ($pairs as $pair) {
+                $nv = explode('=', $pair);
+
+                if ($nv[0] == 'recurring.recurringDetailReference') {
+                    $recurringDetailReference = $nv[1];
+                    break;
+                }
+            }
+
+            if (!empty($recurringDetailReference) &&
+                !empty($paymentResponse['PaymentResult']['PaymentInstrumentData']['CardData'])
```

Comment:
be careful because if any of the higher level keys in ```$paymentResponse['PaymentResult']['PaymentInstrumentData']['CardData']``` doesn't exist (PaymentResult or PaymentInstrumentData) then it will generate a Notice or a Warning. In this case use it with isset.